### ISSUE TITLE
Edit Multimodal Large Language Models

### ISSUE THREAD

**Issue by haohaodw** [zjunlp/EasyEdit#88]
Hello, first of all, congratulations on your work (Can We Edit Multimodal Large Language Models?) being accepted by EMNLP20223. I am very interested in your article, but I encountered some problems when reading the code. I hope to get your answers.

There is a line of code in the evaluation code: "pred = pred[:, :-1]"(in evaluate.py  Function : compute_multimodal_edit_quality, about line 430), why do you delete the last predicted value ? And what is the role of the subsequent line of  code: # targ = targ[:, 1:] ((in evaluate.py  Function : compute_multimodal_edit_quality, about line 431). Although this line of code has been canceled, is this line of code needed in some cases?

**Comment by tbozhong** [comment]
Thanks for your attention!

As we all know, in decoder-only architectures like OPT, GPT, and Llama, the design is centered on next token prediction. In our specific implementation, this implies that the final token does not form part of either the input or the output. Consequently, we remove the last token using `pred = pred[:, :-1]`. The line `# targ = targ[:, 1:]` is displayed because we tokenize the label without adding a special token, so it's not necessary to remove the first token in our code.

**Comment by haohaodw** [comment]
Thank you very much for your reply. Your reply answered my confusion very well. But with in-depth study, I have two new questions.
Question 1: Are the inputs to the model the same in the evaluate stage and in the train stage? Are questions and answers concatenated as input?
Question 2: What is the relationship between the training set and the test set in the data set? Isn't edit success tested on the question you edited (that is, the training set)?

**Comment by cheng-siyuan** [comment]
For Question 1: Yes, for the input it is necessary to concatenate questions and answers.
For Question 2: In the editing task, the training set could be used in Meta-Learning's editing method for training the editor. This is to allow the model to have the ability to adjust its own parameters according to the editing target. For some methods that do not require training, we can directly use the validation set. Note that all metrics should be obtained on the validation set.

**Comment by haohaodw** [comment]
Thank you very much for your reply. In other words, not all methods need to be trained on the training set. Methods such as Grace (https://arxiv.org/abs/2211.11031) can be edited and tested directly on the test set. I don't know if my understanding is correct.

I was also confused by another question. I noticed that you chose the QK-VQA data set when evaluating the model MultiModal Locality, and the index on the base model reached 100, which means that the base model can fully answer all questions. But I called Blip2-opt2.7b on huggingface to test this result, and found that many questions were answered incorrectly by Base model.


Here is my test code：
    import requests
    from PIL import Image
    from transformers import Blip2Processor, Blip2ForConditionalGeneration

    processor_test = Blip2Processor.from_pretrained("Salesforce/blip2-opt-2.7b")
    model_test = Blip2ForConditionalGeneration.from_pretrained("Salesforce/blip2-opt-2.7b", device_map="auto")

    for i, batch in tqdm(enumerate(eval_loader)):
        pixel_values=batch["loc_image"]["image"]
        labels=batch["loc_image"]["labels"]
        input_ids=batch["loc_image"]["prompt_ids"]
        outputs = model_test.generate(input_ids=input_ids, pixel_values=pixel_values)
        re=processor_test.batch_decode(outputs, skip_special_tokens=True)
        ori=processor_test.batch_decode(labels, skip_special_tokens=True)

#####batch["loc_image"]["prompt_ids"] is the tokenized id of prompt, such as 'Question: How fast can you you be going? Short answer: '.

**Comment by tbozhong** [comment]
Indeed, not all methods need to be trained on the training set.

Regarding MultiModal Locality, I recommend referring to our paper on this metric for a comprehensive understanding. Essentially, The locality metric is evaluated based on the rate at which the post-edit model’s predictions remain unchanged compared to the pre-edit model. It's important to note that this does not imply the base model can answer all questions comprehensively. The assessment focuses on the consistency of the model's responses before and after edits, rather than its overall knowledge or answer completeness.

**Comment by zxlzr** [comment]
Do you have any further questions?

**Comment by haohaodw** [comment]
Thank you very much for your patient answer. Your reply inspired me a lot and solved my problem. I have no other questions for now.
